### PR TITLE
Improve documentation following release failure

### PR DESCRIPTION
**Pull request by JoelClemence**
* Correct existing documentation to be correct not referencing tilt etc
* Add documentation of enviroment variables
* Provide brief description of the user journey
* Describe some of the notable frameworks adopted and their purposes - not gotcha around `joi`
* Add some guidance around some steps for working locally

### CODE REVIEW

**Review by sbasile-ch — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,24 +1,79 @@
...
+7. Check answers
+8. User submits and sees the Update submitted message
+
+## Notable Third Party Frameworks used
+
+- `express` for http server/routing
+- `joi` for body validation
+
+  If modifying the body being posted to `/change-email-address` the `joi`
+  schema **must** be updated otherwise requests will fail. and the errors may
```
Comment (ch-tstrickland) [inline README.md:59]: fail. and

File: README.md [hunk 2]
```diff
@@ -44,15 +100,87 @@ and configure your IDE accordingly. For VS-CODE, the following can be added to .
...
+  ```
 
 These instructions are for a local docker environment.
 
-### Running Tests
+### Running unit tests
+
 In order to run tests locally you will need to do the following:
+
 1. Navigate to /registered-email-address-web/
```
Comment (ch-tstrickland) [inline README.md:111]: They're using git submodules??

File: README.md [hunk 3]
```diff
@@ -1,24 +1,79 @@
...
+- `joi` for body validation
+
+  If modifying the body being posted to `/change-email-address` the `joi`
+  schema **must** be updated otherwise requests will fail. and the errors may
+  be quite opaque.
+- `govuk-frontend` etc. for providing Nunjucks components used to compose
+  the screens.
+
+## Developing the service locally
+
```
Comment (ch-tstrickland) [inline README.md:65]: Prerequisites: a populated CHIPS database. You will need to know the user schema for this - it takes the form `CAPDEVXX2` for example `CAPDEVJS2`

If you have not already got such a populated CHIPS db, you will need to request a schema. See https://github.com/companieshouse/docker-chs-development/blob/master/docs/chips.md

File: README.md (hunk 2 above)
Comment (nevans4-ch) [reply README.md:111]: Lots of our repos use this submodule: https://github.com/search?p=1&q=submodule+api-enumerations+org%3Acompanieshouse&type=Code

Comment (JoelClemence) [reply README.md:111]: Yep indeedy - `api-enumerations`

[the remark this answers is not in the document]
Comment (JoelClemence) [reply README.md]: Clarified further.

**Review by ch-tstrickland — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ch-tstrickland) [reply README.md]: I'd forgotten about this. Still not sure that git submodules are a good idea usually, but I lack context.

**Pull request merged**